pilot/cmd/pilot-agent/status/ready/probe.go

```diff
@@ -44,13 +44,31 @@ func (p *Probe) Check() error {
 
 	// Envoy has received some configuration, make sure that configuration has been received for
 	// all inbound ports.
-	if err := p.checkInboundConfigured(); err != nil {
+	if err := p.checkInbound(); err != nil {
 		return err
 	}
 
 	return p.checkServerInfo()
 }
 
+func (p *Probe) checkInbound() error {
+	if !p.ProbeByVirtualInboundListener || p.NodeType != model.SidecarProxy {
+		return p.checkInboundConfigured()
+	}
+	return p.checkInboundVirtualListener()
+
+}
+
+func (p *Probe) checkInboundVirtualListener() error {
+	// Warning: this is not reliable at envoy start up.
+	// TODO(silentdai): switch to /configdump
+	err := util.HasListenerName(p.LocalHostAddr, p.AdminPort, networking.VirtualInboundListenerName)
```

Comment:
is having the 15006 listener sufficient, or do we want to check the filter chains to get the same functionality as before?

Reply:
Previously on sidecar proxy we check if one among the X `ip:containerPort` inbound listener is active, not all of them.
Since the goal is to deprecate containerPort, check 15006 is probably the best we can do. 

Along with SidecarScope I think this implementation may not good but we can deal with it later

Reply:
Yes but before we would send initial config, then readiness fails since it doesn't fail inbound listeners, then it gets one inbound listener and we assume that means it got all and succeed. Now we get the initial config, which always contains 15006 listener.

Really this check does nothing, because if we ever send any LDS, then we know it has 15006, and we already check we got a CDS and LDS sent

Reply:
I didn't consider the possibility that there could be a moment that a virtual listener could arrive before the other inbound listeners. 
I am not sure if it's the readiness probe that resolve that issue under the background that `containerPort` are going to be deprecated.

Reply:
yes it's the whole point of this readiness probe afaik. I think @nmittler added this.

so if we do this then we should just disable the check I don't think there is any point looking for the 15006 listener

Reply:
@howardjohn I don't get your point. The readiness probe is checking if 15006 is active. With this check we expect

- Envoy receives the LDS, expecting all the listener config is received
- 15006 inbound listener is **active** which means the dependencies of are initialized. Notes the dependencies includes SDS which may be time consuming at certain loads, which is more than CDS. 

AFAIK, this is more than a LDS sent counter since I don't think LDS succeed counter bump after the listener turned active.